pkg/utils/authcheck/fake.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package authcheck provides utilities to check authentication configuration for data plane resources.
+// File authcheck contains functions to run customized checks inside of a Pod.
+package authcheck
+
+import (
+	"context"
+	"errors"
+)
+
+type FakeAuthenticationCheck struct {
+	authType AuthType
+	noError  bool
+}
+
+func NewFakeAuthenticationCheck(authType AuthType, noError bool) AuthenticationCheck {
+	return &FakeAuthenticationCheck{
+		authType: authType,
+		noError:  noError,
+	}
+}
```

Comment:
You don't have to expose such a func; instead you can expose `authType` and `noError`.